api/views.py

```diff
@@ -353,8 +353,15 @@ def __get_key(item):
 
         return response.Response(data)
 
-    @transaction.atomic
     def create(self, request, *args, **kwargs):
+        while True:
+            try:
+                return self.__create__(request, *args, **kwargs)
+            except IntegrityError:
```

Comment:
I don't think we can allow _any_ `IntegrityError` here. A duplicated test job id is also going to raise an `IntegrityError`, and since that will never be solved by just retrying, we will have an infinite loop here.